File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        check_list = []
        result = []

        for idx,num in enumerate(nums):
            #idx gets 0
            check_list = nums.copy()
            check_list.pop(idx)
            #check_list = [3]

            for i in check_list:
                if (num+i) == target:
                    
                    result = [idx,nums.index(i)]

        return result

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int size = nums.size();
        unordered_map<int, int> freq;
        
        for(int i = 0; i < size; i++) {
            if(freq.contains(target - nums[i])) {
                return {i, freq[target - nums[i]]};
            }
            freq[nums[i]] = i;
        }
        return {-1,-1};        
    }
};